homework-submission/week2/app.js

```diff
@@ -0,0 +1,101 @@
+class TodoListItem extends React.Component {
+  render() {
+    const { id, description, deadline, done, ClickHandler } = this.props;
+    return (
+      <li key={id} className={done ? 'done' : ''} onClick={ClickHandler}>
+        {description}, {deadline}
+      </li>
+    );
+  }
+}
+
+const DynamicList = ({ listArray, removeTodoItem }) => {
+  return (
+    <div className="dynamic_list">
+      <h1>Dynamic List</h1>
+      <ul>
+        {listArray.map(listItem => {
+          return (
+            <TodoListItem
+              key={listItem.id}
+              onClick={removeTodoItem}
+              description={listItem.description}
+              deadline={listItem.deadline}
+              done={listItem.done}
+            />
+          );
+        })}
+      </ul>
+    </div>
+  );
+};
+
+const TodoItemForm = ({ addingTodo }) => {
+  return (
+    <form className="form" onSubmit={addingTodo}>
```

Comment:
instead of letting user put in an id, you can generate it while adding todo to state